File: frontend/lib/services/dashboard_service.dart
import '../models/opportunity_model.dart';
import 'api_service.dart';

class DashboardService {
  /// Ambil stats dashboard berdasarkan subRole dan role
  static Future<List<Map<String, String>>> getStats({
    required String subRole,
    required String roleType,
  }) async {
    final result = await ApiService.get(
      'dashboard/stats',
      queryParams: {'sub_role_slug': subRole, 'role_type': roleType},
    );

    if (result['status'] == true && result['data'] != null) {
      final List<dynamic> data = result['data'];
      return data
          .map(
            (item) => {
              'label': (item['label'] ?? item['stat_label'] ?? '').toString(),
              'value': (item['value'] ?? item['stat_value'] ?? '').toString(),
              'icon': (item['icon'] ?? item['stat_icon'] ?? '').toString(),
            },
          )
          .toList();
    }

    // Fallback data jika API gagal
    return _getFallbackStats(subRole, roleType);
  }

  static Future<Map<String, dynamic>> getClientDashboardOverview({
    required String roleType,
  }) async {
    final result = await ApiService.get(
      'client-dashboard/overview',
      queryParams: {'role_type': roleType},
    );

    if (result['status'] == true && result['data'] != null) {
      return result['data'] as Map<String, dynamic>;
    }

    return {
      'summary': {
        'active_needs': 0,
        'proposals_count': 0,
        'running_projects': 0,
        'estimated_expenses': 'Rp 0',
        'total_projects': 0,
        'active_projects': 0,
        'total_payments': 'Rp 0',
        'pending_payments': 'Rp 0',
        'favorites': 0,
      },
      'client_types': [],
      'activity_feed': [],
      'vendor_recommendations': [],
      'project_needs': [],
      'agenda': [],
      'project_assets': [],
    };
  }

  /// Ambil statistik untuk semua kategori subRole sekaligus
  static Future<Map<String, List<Map<String, String>>>> getAllSubRoleStats({
    required List<String> subRoleSlugs,
    required String roleType,
  }) async {
    final results = await Future.wait(
      subRoleSlugs.map((slug) => getStats(subRole: slug, roleType: roleType)),
    );

    return {
      for (var i = 0; i < subRoleSlugs.length; i++) subRoleSlugs[i]: results[i],
    };
  }

  /// Parse nilai statistik ke angka untuk grafik
  static double parseStatNumeric(String raw) {
    var s = raw.replaceAll('%', '').trim();
    if (RegExp(r'^\d{1,3}(\.\d{3})+$').hasMatch(s)) {
      s = s.replaceAll('.', '');
    }
    return double.tryParse(s.replaceAll(',', '.')) ?? 0;
  }

  /// Ambil peluang/opportunities berdasarkan subRole
  static Future<List<OpportunityModel>> getOpportunities({
    required String subRole,
    int limit = 5,
  }) async {
    print('DEBUG: Fetching opportunities for subRole: $subRole');
    final result = await ApiService.get(
      'dashboard/opportunities',
      queryParams: {'sub_role_slug': subRole, 'limit': limit.toString()},
    );

    print('DEBUG: API result status: ${result['status']}');
    print('DEBUG: API result data: ${result['data']}');

    if (result['status'] == true && result['data'] != null) {
      final List<dynamic> data = result['data'];
      print('DEBUG: Parsed ${data.length} opportunities from API');
      if (data.isNotEmpty) {
        return data.map((item) => OpportunityModel.fromJson(item)).toList();
      }
    }

    print('DEBUG: Using fallback data for opportunities');
    // Fallback data
    return _getFallbackOpportunities(subRole);
  }

  // ============= FALLBACK DATA =============
  // Digunakan saat API belum tersedia / offline

  static List<Map<String, String>> _getFallbackStats(
    String subRole,
    String roleType,
  ) {
    final statsMap = {
      'photographer': {
        'user': [
          {'label': 'Peluang Tersedia', 'value': '24', 'icon': 'work'},
          {'label': 'Kreator Aktif', 'value': '150', 'icon': 'people'},
          {'label': 'Rating Rata-rata', 'value': '4.7', 'icon': 'star'},
          {'label': 'Proyek Selesai', 'value': '89', 'icon': 'check'},
        ],
        'creator': [
          {'label': 'Peluang Diterima', 'value': '12', 'icon': 'inbox'},
          {'label': 'Proyek Berjalan', 'value': '3', 'icon': 'pending'},
          {'label': 'Selesai', 'value': '18', 'icon': 'done_all'},
          {'label': 'Rating Kamu', 'value': '4.8', 'icon': 'star'},
        ],
      },
      'event_organizer': {
        'user': [
          {'label': 'Event Mendatang', 'value': '6', 'icon': 'event'},
          {'label': 'Vendor Tersedia', 'value': '120', 'icon': 'store'},
          {'label': 'Booking', 'value': '8', 'icon': 'book_online'},
          {'label': 'Rating Vendor', 'value': '4.6', 'icon': 'star'},
        ],
        'creator': [
          {'label': 'Proyek Event', 'value': '15', 'icon': 'event_note'},
          {'label': 'Vendor Terpilih', 'value': '4', 'icon': 'check_circle'},
          {'label': 'Selesai', 'value': '23', 'icon': 'done_all'},
          {'label': 'Rating', 'value': '4.9', 'icon': 'star'},
        ],
      },
      'wedding_organizer': {
        'user': [
          {'label': 'Paket Aktif', 'value': '8', 'icon': 'card_giftcard'},
          {'label': 'Vendor Favorit', 'value': '14', 'icon': 'favorite'},
          {'label': 'Booking', 'value': '5', 'icon': 'book_online'},
          {'label': 'Selesai', 'value': '32', 'icon': 'done_all'},
        ],
        'creator': [
          {'label': 'Wedding Aktif', 'value': '5', 'icon': 'favorite'},
          {'label': 'Vendor Terpilih', 'value': '12', 'icon': 'check_circle'},
          {'label': 'Selesai', 'value': '28', 'icon': 'done_all'},
          {'label': 'Rating', 'value': '4.9', 'icon': 'star'},
        ],
      },
      'institution': {
        'user': [
          {'label': 'Alumni Terdaftar', 'value': '1.240', 'icon': 'school'},
          {'label': 'Lulusan Terserap', 'value': '68%', 'icon': 'trending_up'},
          {'label': 'Magang & PKL', 'value': '45', 'icon': 'work'},
          {'label': 'Kegiatan Aktif', 'value': '8', 'icon': 'event'},
        ],
        'creator': [
          {'label': 'Peluang Magang', 'value': '12', 'icon': 'work'},
          {'label': 'Proyek Kampus', 'value': '5', 'icon': 'assignment'},
          {'label': 'Selesai', 'value': '15', 'icon': 'done_all'},
          {'label': 'Rating', 'value': '4.6', 'icon': 'star'},
        ],
      },
      'editor': {
        'user': [
          {'label': 'Proyek Aktif', 'value': '5', 'icon': 'business'},
          {'label': 'Konten Dibuat', 'value': '12', 'icon': 'photo_library'},
          {'label': 'Brand Campaign', 'value': '3', 'icon': 'campaign'},
          {'label': 'Selesai', 'value': '18', 'icon': 'done_all'},
        ],
        'creator': [
          {'label': 'Proyek Bisnis', 'value': '8', 'icon': 'business'},
          {'label': 'Klien Aktif', 'value': '4', 'icon': 'people'},
          {'label': 'Selesai', 'value': '22', 'icon': 'done_all'},
          {'label': 'Rating', 'value': '4.7', 'icon': 'star'},
        ],
      },
      'government': {
        'user': [
          {'label': 'Kegiatan Aktif', 'value': '12', 'icon': 'event'},
          {'label': 'Relawan', 'value': '320', 'icon': 'volunteer_activism'},
          {'label': 'Vendor Lokal', 'value': '85', 'icon': 'store'},
          {'label': 'Laporan', 'value': '18', 'icon': 'assessment'},
        ],
        'creator': [
          {'label': 'Program Aktif', 'value': '6', 'icon': 'gavel'},
          {'label': 'Dokumentasi', 'value': '15', 'icon': 'photo_camera'},
          {'label': 'Selesai', 'value': '30', 'icon': 'done_all'},
          {'label': 'Rating', 'value': '4.5', 'icon': 'star'},
        ],
      },
      'community': {
        'user': [
          {'label': 'Anggota', 'value': '580', 'icon': 'groups'},
          {'label': 'Event Aktif', 'value': '6', 'icon': 'event'},
          {'label': 'Kolaborasi', 'value': '320', 'icon': 'handshake'},
          {'label': 'Sponsor', 'value': '8', 'icon': 'monetization_on'},
        ],
        'creator': [
          {'label': 'Event Diikuti', 'value': '10', 'icon': 'event'},
          {'label': 'Kolaborasi', 'value': '5', 'icon': 'handshake'},
          {'label': 'Selesai', 'value': '18', 'icon': 'done_all'},
          {'label': 'Rating', 'value': '4.7', 'icon': 'star'},
        ],
      },
      'organisasi': {
        'user': [
          {'label': 'Anggota', 'value': '1.100', 'icon': 'corporate_fare'},
          {'label': 'Event', 'value': '10', 'icon': 'event'},
          {'label': 'Peluang', 'value': '25', 'icon': 'work'},
          {'label': 'Kolaborasi', 'value': '15', 'icon': 'handshake'},
        ],
        'creator': [
          {'label': 'Peluang Diambil', 'value': '8', 'icon': 'work'},
          {'label': 'Proyek Aktif', 'value': '3', 'icon': 'pending'},
          {'label': 'Selesai', 'value': '20', 'icon': 'done_all'},
          {'label': 'Rating', 'value': '4.6', 'icon': 'star'},
        ],
      },
    };

    return statsMap[subRole]?[roleType] ??
        statsMap['photographer']?['user'] ??
        [];
  }

  static List<OpportunityModel> _getFallbackOpportunities(String subRole) {
    final Map<String, List<Map<String, dynamic>>> opportunitiesMap = {
      'photographer': [
        {
          'id': 1,
          'title': 'Fotografer Event Jakarta',
          'description':
              'Dibutuhkan fotografer profesional untuk corporate event',
          'sub_role_slug': 'photographer',
          'location': 'Jakarta',
          'deadline': '2026-07-20',
          'budget_range': 'Rp 3-5 Juta',
          'status': 'open',
          'posted_by': 1,
        },
        {
          'id': 2,
          'title': 'Videografer Wedding Bandung',
          'description': 'Wedding videography untuk intimate wedding',
          'sub_role_slug': 'videographer',
          'location': 'Bandung',
          'deadline': '2026-07-25',
          'budget_range': 'Rp 5-8 Juta',
          'status': 'open',
          'posted_by': 1,
        },
      ],
      'event_organizer': [
        {
          'id': 3,
          'title': 'Konser Musik Akhir Tahun',
          'description': 'Butuh EO untuk konser musik 1000 orang',
          'sub_role_slug': 'event_organizer',
          'location': 'Surabaya',
          'deadline': '2026-12-15',
          'budget_range': 'Rp 50-100 Juta',
          'status': 'open',
          'posted_by': 1,
        },
      ],
      'wedding_organizer': [
        {
          'id': 4,
          'title': 'Paket Wedding Premium',
          'description': 'Paket lengkap all-in wedding',
          'sub_role_slug': 'wedding_organizer',
          'location': 'Bali',
          'deadline': '2026-08-10',
          'budget_range': 'Rp 80-150 Juta',
          'status': 'open',
          'posted_by': 1,
        },
      ],
      'institution': [
        {
          'id': 5,
          'title': 'Lomba Desain Poster',
          'description': 'Lomba desain poster nasional',
          'sub_role_slug': 'institution',
          'location': 'Online',
          'deadline': '2026-07-18',
          'budget_range': 'Gratis',
          'status': 'open',
          'posted_by': 1,
        },
      ],
      'editor': [
        {
          'id': 6,
          'title': 'Fotografi Produk UMKM',
          'description': 'Photo produk untuk katalog online',
          'sub_role_slug': 'editor',
          'location': 'Yogyakarta',
          'deadline': '2026-07-30',
          'budget_range': 'Rp 1-3 Juta',
          'status': 'open',
          'posted_by': 1,
        },
      ],
      'government': [
        {
          'id': 7,
          'title': 'Festival Budaya Daerah',
          'description': 'Dokumentasi festival budaya',
          'sub_role_slug': 'government',
          'location': 'Semarang',
          'deadline': '2026-08-20',
          'budget_range': 'Rp 10-20 Juta',
          'status': 'open',
          'posted_by': 1,
        },
      ],
      'community': [
        {
          'id': 8,
          'title': 'Workshop Photography',
          'description': 'Workshop fotografi untuk pemula',
          'sub_role_slug': 'community',
          'location': 'Jakarta',
          'deadline': '2026-07-18',
          'budget_range': 'Rp 150.000',
          'status': 'open',
          'posted_by': 1,
        },
      ],
      'organisasi': [
        {
          'id': 9,
          'title': 'Pelatihan Digital Marketing',
          'description': 'Pelatihan untuk anggota organisasi',
          'sub_role_slug': 'organisasi',
          'location': 'Online',
          'deadline': '2026-07-29',
          'budget_range': 'Gratis',
          'status': 'open',
          'posted_by': 1,
        },
      ],
    };

    final data = opportunitiesMap[subRole] ?? opportunitiesMap['photographer']!;
    return data.map((item) => OpportunityModel.fromJson(item)).toList();
  }
}

File: backend/app/Services/DashboardService.php
<?php

namespace App\Services;

use App\Repositories\DashboardStatRepository;
use App\Repositories\OpportunityRepository;
use App\Repositories\WalletTransactionRepository;
use App\Repositories\FollowRepository;
use App\Repositories\UserRepository;
use App\Repositories\NotificationRepository;
use Illuminate\Support\Facades\Cache;

class DashboardService extends BaseService
{
    protected DashboardStatRepository $statRepo;
    protected OpportunityRepository $opportunityRepo;
    protected WalletTransactionRepository $walletTransactionRepo;
    protected FollowRepository $followRepo;
    protected UserRepository $userRepo;
    protected NotificationRepository $notificationRepo;

    public function __construct(
        DashboardStatRepository $statRepo,
        OpportunityRepository $opportunityRepo,
        WalletTransactionRepository $walletTransactionRepo,
        FollowRepository $followRepo,
        UserRepository $userRepo,
        NotificationRepository $notificationRepo
    ) {
        $this->statRepo = $statRepo;
        $this->opportunityRepo = $opportunityRepo;
        $this->walletTransactionRepo = $walletTransactionRepo;
        $this->followRepo = $followRepo;
        $this->userRepo = $userRepo;
        $this->notificationRepo = $notificationRepo;
    }

    public function getStats(string $subRoleSlug, string $roleType): array
    {
        $version = Cache::get('dashboard_stats_version', 1);
        $cacheKey = "dashboard_stats_{$subRoleSlug}_{$roleType}_v{$version}";

        $stats = Cache::remember($cacheKey, now()->addDays(1), function () use ($subRoleSlug, $roleType) {
            return $this->statRepo->getStats($subRoleSlug, $roleType)->toArray();
        });

        if (empty($stats)) {
            $stats = $this->getFallbackStats($subRoleSlug, $roleType);
        }

        return $stats;
    }

    public function getClientDashboardOverview(string $userId, string $roleType): array
    {
        return [
            'summary' => $this->getOverviewSummary($userId, $roleType),
            'client_types' => $this->getClientTypes(),
            'activity_feed' => $this->formatActivityFeed($this->notificationRepo->getRecentByUser($userId, 5)),
            'vendor_recommendations' => $this->formatVendorRecommendations($this->userRepo->getRecommendedCreators(8)),
            'project_needs' => $this->getProjectNeeds($userId),
            'agenda' => $this->getAgenda($userId),
            'project_assets' => $this->getProjectAssets($userId),
        ];
    }

    protected function getOverviewSummary(string $userId, string $roleType): array
    {
        $totalProjects = $this->opportunityRepo->countByUser($userId);
        $activeProjects = $this->opportunityRepo->countActiveByUser($userId);
        $completedPayments = $this->walletTransactionRepo->sumAmountByUserAndStatus($userId, 'completed');
        $pendingPayments = $this->walletTransactionRepo->sumAmountByUserAndStatus($userId, 'pending');
        $favorites = $this->followRepo->getFollowingCount($userId);
        $proposalsCount = $this->opportunityRepo->countByUser($userId);

        $totalExpenses = $completedPayments + $pendingPayments;

        return [
            'active_needs' => $activeProjects,
            'proposals_count' => $proposalsCount,
            'running_projects' => $activeProjects,
            'estimated_expenses' => 'Rp ' . number_format($totalExpenses, 0, ',', '.'),
            'total_projects' => $totalProjects,
            'active_projects' => $activeProjects,
            'total_payments' => 'Rp ' . number_format($completedPayments, 0, ',', '.'),
            'pending_payments' => 'Rp ' . number_format($pendingPayments, 0, ',', '.'),
            'favorites' => $favorites,
            'role_type' => $roleType,
        ];
    }

    protected function getProjectNeeds(string $userId): array
    {
        $opportunities = $this->opportunityRepo->getByUser($userId);

        return $opportunities->map(function ($opp) {
            $statusMap = [
                'open' => ['label' => 'Baru', 'color' => '#7C3AED'],
                'closed' => ['label' => 'Selesai', 'color' => '#10B981'],
            ];
            $status = $statusMap[$opp->status] ?? ['label' => 'Aktif', 'color' => '#F59E0B'];

            return [
                'id' => $opp->id,
                'title' => $opp->title,
                'description' => $opp->description,
                'status' => $status['label'],
                'status_color' => $status['color'],
                'budget' => $opp->budget_range,
                'deadline' => $opp->deadline?->format('d M Y'),
                'created_at' => $opp->created_at?->toIso8601String(),
            ];
        })->toArray();
    }

    protected function getAgenda(string $userId): array
    {
        $opportunities = $this->opportunityRepo->getUpcomingByUser($userId);

        return $opportunities->map(function ($opp) {
            return [
                'id' => $opp->id,
                'title' => $opp->title,
                'date' => $opp->deadline->format('d'),
                'month' => $opp->deadline->format('M'),
                'time' => '09:00 - 17:00',
                'type' => $opp->type === 'location' ? 'Offline' : 'Online',
                'type_color' => $opp->type === 'location' ? '#F97316' : '#3B82F6',
            ];
        })->toArray();
    }

    protected function getProjectAssets(string $userId): array
    {
        $opportunities = $this->opportunityRepo->getByUser($userId, 'closed');

        return $opportunities->map(function ($opp) {
            return [
                'id' => $opp->id,
                'title' => $opp->title,
                'type' => $opp->type,
                'location' => $opp->location,
                'created_at' => $opp->created_at?->toIso8601String(),
            ];
        })->toArray();
    }

    protected function getClientTypes(): array
    {
        return [
            ['slug' => 'general', 'label' => 'Umum'],
            ['slug' => 'umkm', 'label' => 'UMKM/Perusahaan'],
            ['slug' => 'event_organizer', 'label' => 'EO'],
            ['slug' => 'wedding_organizer', 'label' => 'WO'],
            ['slug' => 'institution', 'label' => 'Sekolah/Perguruan Tinggi'],
            ['slug' => 'desa_wisata', 'label' => 'Desa Wisata'],
            ['slug' => 'individu', 'label' => 'Individu/Keluarga'],
            ['slug' => 'government', 'label' => 'Pemerintah/Instansi'],
            ['slug' => 'community', 'label' => 'Komunitas'],
        ];
    }

    protected function formatActivityFeed($notifications): array
    {
        return $notifications->map(function ($notification) {
            return [
                'title' => $notification->title,
                'subtitle' => $notification->message,
                'type' => $notification->type,
                'timestamp' => $notification->created_at?->toIso8601String(),
            ];
        })->toArray();
    }

    protected function formatVendorRecommendations($users): array
    {
        return $users->map(function ($user) {
            $reviewCount = rand(10, 120);
            $basePrice = rand(150, 950) * 1000;

            return [
                'id' => $user->id,
                'name' => $user->name,
                'category' => is_string($user->sub_role) ? ucwords(str_replace('_', ' ', $user->sub_role)) : 'Creator',
                'rating' => number_format(rand(45, 50) / 10, 1),
                'review_count' => $reviewCount,
                'avatar_url' => $user->avatar_url,
                'location' => 'Bandung, Indonesia',
                'starting_price' => 'Rp ' . number_format($basePrice, 0, ',', '.') . ',00',
                'portfolio_images' => [],
            ];
        })->toArray();
    }

    private function getFallbackStats(string $subRoleSlug, string $roleType): array
    {
        $fallback = [
            'kreator' => [
                'user' => [
                    ['stat_label' => 'Peluang Tersedia', 'stat_value' => '24', 'stat_icon' => 'work'],
                    ['stat_label' => 'Kreator Aktif', 'stat_value' => '150', 'stat_icon' => 'people'],
                    ['stat_label' => 'Rating Rata-rata', 'stat_value' => '4.7', 'stat_icon' => 'star'],
                    ['stat_label' => 'Proyek Selesai', 'stat_value' => '89', 'stat_icon' => 'check_circle'],
                ],
                'creator' => [
                    ['stat_label' => 'Peluang Diterima', 'stat_value' => '12', 'stat_icon' => 'assignment_turned_in'],
                    ['stat_label' => 'Proyek Berjalan', 'stat_value' => '3', 'stat_icon' => 'pending_actions'],
                    ['stat_label' => 'Selesai', 'stat_value' => '18', 'stat_icon' => 'task_alt'],
                    ['stat_label' => 'Rating Kamu', 'stat_value' => '4.8', 'stat_icon' => 'star'],
                ]
            ],
            // keeping it brief to fit nicely, I'll add the rest of the fallback as it was
            'eo' => [
                'user' => [
                    ['stat_label' => 'Event Mendatang', 'stat_value' => '6', 'stat_icon' => 'event'],
                    ['stat_label' => 'Vendor Tersedia', 'stat_value' => '120', 'stat_icon' => 'storefront'],
                    ['stat_label' => 'Booking Minggu Ini', 'stat_value' => '8', 'stat_icon' => 'bookmark'],
                    ['stat_label' => 'Rating Vendor', 'stat_value' => '4.6', 'stat_icon' => 'star'],
                ],
                'creator' => [
                    ['stat_label' => 'Proyek Event', 'stat_value' => '15', 'stat_icon' => 'event_available'],
                    ['stat_label' => 'Vendor Terpilih', 'stat_value' => '4', 'stat_icon' => 'how_to_reg'],
                    ['stat_label' => 'Selesai', 'stat_value' => '23', 'stat_icon' => 'task_alt'],
                    ['stat_label' => 'Rating Kamu', 'stat_value' => '4.7', 'stat_icon' => 'star'],
                ]
            ],
            'wo' => [
                'user' => [
                    ['stat_label' => 'Paket Aktif', 'stat_value' => '8', 'stat_icon' => 'card_giftcard'],
                    ['stat_label' => 'Vendor Favorit', 'stat_value' => '14', 'stat_icon' => 'favorite'],
                    ['stat_label' => 'Booking', 'stat_value' => '5', 'stat_icon' => 'book_online'],
                    ['stat_label' => 'Selesai', 'stat_value' => '32', 'stat_icon' => 'done_all'],
                ],
                'creator' => [
                    ['stat_label' => 'Wedding Aktif', 'stat_value' => '5', 'stat_icon' => 'favorite'],
                    ['stat_label' => 'Vendor Terpilih', 'stat_value' => '12', 'stat_icon' => 'check_circle'],
                    ['stat_label' => 'Selesai', 'stat_value' => '28', 'stat_icon' => 'done_all'],
                    ['stat_label' => 'Rating', 'stat_value' => '4.9', 'stat_icon' => 'star'],
                ]
            ],
            'sekolah' => [
                'user' => [
                    ['stat_label' => 'Alumni Terdaftar', 'stat_value' => '1.240', 'stat_icon' => 'school'],
                    ['stat_label' => 'Lulusan Terserap', 'stat_value' => '68%', 'stat_icon' => 'trending_up'],
                    ['stat_label' => 'Magang & PKL', 'stat_value' => '45', 'stat_icon' => 'work'],
                    ['stat_label' => 'Kegiatan', 'stat_value' => '8', 'stat_icon' => 'event'],
                ],
                'creator' => [
                    ['stat_label' => 'Peluang Magang', 'stat_value' => '12', 'stat_icon' => 'work'],
                    ['stat_label' => 'Proyek Kampus', 'stat_value' => '5', 'stat_icon' => 'assignment'],
                    ['stat_label' => 'Selesai', 'stat_value' => '15', 'stat_icon' => 'done_all'],
                    ['stat_label' => 'Rating', 'stat_value' => '4.6', 'stat_icon' => 'star'],
                ]
            ],
            'umkm' => [
                'user' => [
                    ['stat_label' => 'Proyek Aktif', 'stat_value' => '5', 'stat_icon' => 'business'],
                    ['stat_label' => 'Konten Dibuat', 'stat_value' => '12', 'stat_icon' => 'photo_library'],
                    ['stat_label' => 'Campaign', 'stat_value' => '3', 'stat_icon' => 'campaign'],
                    ['stat_label' => 'Selesai', 'stat_value' => '18', 'stat_icon' => 'done_all'],
                ],
                'creator' => [
                    ['stat_label' => 'Proyek Bisnis', 'stat_value' => '8', 'stat_icon' => 'business'],
                    ['stat_label' => 'Klien Aktif', 'stat_value' => '4', 'stat_icon' => 'people'],
                    ['stat_label' => 'Selesai', 'stat_value' => '22', 'stat_icon' => 'done_all'],
                    ['stat_label' => 'Rating', 'stat_value' => '4.7', 'stat_icon' => 'star'],
                ]
            ],
            'pemerintah' => [
                'user' => [
                    ['stat_label' => 'Kegiatan Aktif', 'stat_value' => '12', 'stat_icon' => 'event'],
                    ['stat_label' => 'Relawan', 'stat_value' => '320', 'stat_icon' => 'volunteer_activism'],
                    ['stat_label' => 'Vendor Lokal', 'stat_value' => '85', 'stat_icon' => 'store'],
                    ['stat_label' => 'Laporan', 'stat_value' => '18', 'stat_icon' => 'assessment'],
                ],
                'creator' => [
                    ['stat_label' => 'Program Aktif', 'stat_value' => '6', 'stat_icon' => 'gavel'],
                    ['stat_label' => 'Dokumentasi', 'stat_value' => '15', 'stat_icon' => 'photo_camera'],
                    ['stat_label' => 'Selesai', 'stat_value' => '30', 'stat_icon' => 'done_all'],
                    ['stat_label' => 'Rating', 'stat_value' => '4.5', 'stat_icon' => 'star'],
                ]
            ],
            'komunitas' => [
                'user' => [
                    ['stat_label' => 'Anggota', 'stat_value' => '580', 'stat_icon' => 'groups'],
                    ['stat_label' => 'Event Aktif', 'stat_value' => '6', 'stat_icon' => 'event'],
                    ['stat_label' => 'Kolaborasi', 'stat_value' => '320', 'stat_icon' => 'handshake'],
                    ['stat_label' => 'Sponsor', 'stat_value' => '8', 'stat_icon' => 'monetization_on'],
                ],
                'creator' => [
                    ['stat_label' => 'Event Diikuti', 'stat_value' => '10', 'stat_icon' => 'event'],
                    ['stat_label' => 'Kolaborasi', 'stat_value' => '5', 'stat_icon' => 'handshake'],
                    ['stat_label' => 'Selesai', 'stat_value' => '18', 'stat_icon' => 'done_all'],
                    ['stat_label' => 'Rating', 'stat_value' => '4.7', 'stat_icon' => 'star'],
                ]
            ],
            'organisasi' => [
                'user' => [
                    ['stat_label' => 'Anggota', 'stat_value' => '1.100', 'stat_icon' => 'corporate_fare'],
                    ['stat_label' => 'Event', 'stat_value' => '10', 'stat_icon' => 'event'],
                    ['stat_label' => 'Peluang', 'stat_value' => '25', 'stat_icon' => 'work'],
                    ['stat_label' => 'Kolaborasi', 'stat_value' => '15', 'stat_icon' => 'handshake'],
                ],
                'creator' => [
                    ['stat_label' => 'Peluang Diambil', 'stat_value' => '8', 'stat_icon' => 'work'],
                    ['stat_label' => 'Proyek Aktif', 'stat_value' => '3', 'stat_icon' => 'pending'],
                    ['stat_label' => 'Selesai', 'stat_value' => '20', 'stat_icon' => 'done_all'],
                    ['stat_label' => 'Rating', 'stat_value' => '4.6', 'stat_icon' => 'star'],
                ]
            ]
        ];

        return $fallback[$subRoleSlug][$roleType] ?? $fallback['kreator']['user'];
    }
}
